lib/ft817/ft817.cpp

```diff
@@ -399,9 +400,10 @@ void FT817::getBytes(byte count)
 	while (rigCat.available() < 1 && millis() < startTime + 2000) { ; }
 
 	flushBuffer();
-	for (byte i=count; i>0; i--)
+	for (byte i=0; i<count; i++)
 	{
 		buffer[i] = rigCat.read();
+		delay(5);					// needs a delay in here, otherwise the byte sequence read is incorrect

```

Comment:
I discovered this by accident.  I had some print statements in here while debugging, fixed the problems, it worked, then I removed the print statements and the bytes in the buffer got screwed up... added in a small delay and it works.  Slightly worrying.

Reply:
...thinking about it a bit more... I've read some things about SoftwareSerial being slow and sometimes unreliable.  I have seen some read errors in my FT817 Buddy code in the field and I was thinking about trying to use [NeoSWSerial ](https://www.arduino.cc/reference/en/libraries/neoswserial/) instead to see if that improves things.  It might remove the need for this ugly delay... although it might be down to the radio's timing.